packages/react-utilities/src/utils/getNativeElementProps.ts

```diff
@@ -65,3 +65,31 @@ export function getNativeElementProps<TAttributes extends React.HTMLAttributes<a
 
   return getNativeProps(props, allowedPropNames, excludedPropNames);
 }
+
+/**
+ * Splits the native props into ones that go to the 'root' slot, and ones that go to the 'primary' slot.
+ *
+ * This function is only for use with components that have a primary slot other than 'root'.
+ * Most components should use {@link getNativeElementProps} for their root slot if it is the primary slot.
+ *
+ * @returns An object containing the native props for the `root` and `primary` slots.
+ */
+export const getPartitionedNativeProps = ({
+  tagName,
+  props,
+  excludedPropNames,
+}: {
+  /** The element's tag name (e.g. 'div') */
+  tagName: keyof JSX.IntrinsicElements;
+
+  /** The component's props object */
+  props: Pick<React.HTMLAttributes<HTMLElement>, 'style' | 'className'>;
+
+  /** List of native props to exclude from the returned value */
+  excludedPropNames?: string[];
+}) => {
+  return {
+    root: { style: props.style, className: props.className },
+    primary: getNativeElementProps(tagName, props, [...(excludedPropNames ?? []), 'style', 'className']),
```

Comment:
nit, || transpiles down a bit smaller and will have the same (actually a bit better) behavior here
```suggestion
    primary: getNativeElementProps(tagName, props, [...(excludedPropNames || []), 'style', 'className']),
```